ReadThread.py

```diff
@@ -0,0 +1,76 @@
+from typing import Dict
+
+from PyQt5 import QtCore
+from PyQt5.QtCore import pyqtSignal
+import queue
+
+import RadioController
+
+class ReadThread(QtCore.QThread):
+    # sig_received = pyqtSignal(list)  # TODO Review this change with Andrei
+    sig_received = pyqtSignal(object)  # TODO is it safe to generalize
+    sig_print = pyqtSignal(str)
+
+    def __init__(self, connection, parent=None):
+        QtCore.QThread.__init__(self, parent)
+        self.connection = connection
+        self.connection.registerCallback(self._newData)
+
+        self.dataQueue = queue.Queue()
+
+        self.errored = False
+
+    def _newData(self, data):
+        self.dataQueue.put_nowait(data)
+
+    # Function that is run by the thread when it is not blocked
+    def run(self):
+        self.running = True  # TODO Review purpose of this
+        # Loop that attempts a message send and gets+parses data repeatedly
+        while self.running:
```

Comment:
Another artifact but thought it would be relevant to discuss:
What is the point of having this as a variable?
And if it is a field, why declare it here? Is that good practice?

Reply:
Usually that's how you'd terminate a thread. There is supposed to be a stop() function that sets running to false. But we don't ever do that so that code is obsolete.

Yeah it should have been "declared" in init imho. But I'll just remove it entirely.